torch/_dynamo/variables/builder.py

```diff
@@ -2281,6 +2280,11 @@ def handle_traced_output(example_value, tx, proxy, options, subclass_type, targe
         # tensor, the stored example value will update too!)
         example_value = _clone_input(example_value, tx.fake_mode)
         set_example_value(proxy.node, example_value)
+        # We bind the unbacked symints in sizes/trdies of tensor lazily.
+        # So that subgraphs can access the unbacked symbol's proxy in parent graph
+        # when lifting unbacked symbols of input tensors to subgraph inputs.
+        # We do it lazily because the tensor may not be used in subgraphs.
+        tx.output.current_tracer.lazy_bind_unbacked_symbols(example_value, proxy)
```

Comment:
Sorry, I'm still confused. You need some logic to keep the current_tracer.bound_symbols map up-to-date. Is this the function that does it?

My confusion is that this function seems to only handle unbacked symints, so what is actually putting backed symints into current_tracer.bound_symbols?

Reply:
Copying the dicussion offline for record:
```
1. We need to change bound_symbols after create_graph_input for a basic symbol because that symbol previously is not tracked by current tracer (e.g. it could be the shape of a tensor input). After tracking the symbol, child subgraph can look for the parent's proxy in parent's bound_symbols. 
2. we also need change bound_symbols while we're executing the tensor operations: e.g. for nonzero call, we need to record the u0 in bound symbols otherwise when subgraph looks for the u0 in parent, it cannot find the proxy for u0 in parent.
```